chat-plugins/health-fitness.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Heath & Fitness chat-plugin
+ * Pokemon Showdown - http://pokemonshowdown.com/
+ * 
+ * This handles health and fitness daily challenges as well as a daily quote
+ * 
+ * @license MIT license
+ */
+'use strict';
+
+const FS = require('../fs');
+
+const HF_DATA_PATH = 'config/chat-plugins/health-and-fitness.json';
+const MAX_DATA_LENGTH = 500;
+
+class HealthFitness {
+	constructor() {
+		this.hfData = {
+			cardio: '',
+			gym: '',
+			quote: '',
+			__proto__: null,
+		};
+
+		try {
+			this.hfData = require(`../${HF_DATA_PATH}`);
+		} catch (error) {
+			if (error.code !== 'MODULE_NOT_FOUND') throw error;
+		}
+
+		this.saveData = (() => {
+			let writing = false;
+			let writePending = false;
+			return async () => {
+				if (writing) {
+					writePending = true;
+					return;
+				}
+				writing = true;
+
+				await FS(`${HF_DATA_PATH}.0`).write(JSON.stringify(this.hfData));
+				await FS(`${HF_DATA_PATH}.0`).rename(HF_DATA_PATH);
+
+				writing = false;
+				if (writePending) {
+					writePending = false;
+					setImmediate(() => this.saveData());
+				}
+			};
+		})();
+	}
+	setCardio(cardio) {
+		this.hfData.cardio = cardio;
+		this.saveData();
+	}
+	setGym(gym) {
+		this.hfData.gym = gym;
+		this.saveData();
+	}
+	setQuote(quote) {
+		this.hfData.quote = quote;
+		this.saveData();
+	}
+}
+const HF = new HealthFitness();
+
+exports.commands = {
+	hellafit: 'healthfitness',
+	health: 'healthfitness',
+	fitness: 'healthfitness',
+	healthfitness: {
+		'': function (target, room, user) {
+			if (room.id !== 'healthfitness') return this.errorReply("This command can only be used in Health & Fitness.");
+			if (!this.runBroadcast()) return;
+			const cardio = HF.hfData.cardio || '(not set yet)';
+			const gym = HF.hfData.gym || '(not set yet)';
+			const quote = HF.hfData.quote || '(not set yet)';
+			return this.sendReplyBox(
+				Chat.html`<strong>Cardio Challenge:</strong> ${cardio}<br />` +
```

Comment:
Maybe use Chat.parseText for some of these so links are autolinked?

Reply:
I'm not sure that the room is planning on including links in these. I'll confirm this later :p

Reply:
I've used links in the challenges before, usually to explain a more obscure exercise not many people know